dump/plugins/fdb.py

```diff
@@ -0,0 +1,103 @@
+from dump.match_infra import MatchRequest
+from dump.helper import create_template_dict
+from .executor import Executor
+
+
+class Fdb(Executor):
+    """
+    Debug Dump Plugin for FDB Module
+    """
+    ARG_NAME = "Vlan:fdb_entry"
+
+    def __init__(self, match_engine=None):
+        super().__init__(match_engine)
+
+    def get_all_args(self, ns=""):
+        req = MatchRequest(db="STATE_DB", table="FDB_TABLE", key_pattern="*", ns=ns)
+        ret = self.match_engine.fetch(req)
+        fdb_entries = ret["keys"]
+        return [key.split("|")[-1] for key in fdb_entries]
+
+    def execute(self, params):
+        self.ret_temp = create_template_dict(dbs=["APPL_DB", "ASIC_DB", "STATE_DB"])
+        fdb_entry = params[Fdb.ARG_NAME]
+        self.ns = params["namespace"]
+        self.init_fdb_appl_info(fdb_entry)
+        self.init_asic_fdb_info(fdb_entry)
+        self.init_state_fdb_info(fdb_entry)
+        return self.ret_temp
+
+    def add_to_ret_template(self, table, db, keys, err):
+        if not err and keys:
+            self.ret_temp[db]["keys"].extend(keys)
+            return True
+        else:
+            self.ret_temp[db]["tables_not_found"].extend([table])
+            return False
+
+    def init_state_fdb_info(self, fdb_name):
+        req = MatchRequest(db="STATE_DB", table="FDB_TABLE", key_pattern=fdb_name, ns=self.ns)
+        ret = self.match_engine.fetch(req)
+        self.add_to_ret_template(req.table, req.db, ret["keys"], ret["error"])
+
+    def init_fdb_appl_info(self, fdb_name):
```

Comment:
The idea to not add APP_DB tables when missing is on purpose. In most scenarios FDB will be present only in ASIC and STATE_DB. Only in EVPN and MCLAG they are supposed to be programmed through control plane. If we display table not found for every hardware learnt entry it might create a wrong impression as something is missing particularly in fdb scenario.

Reply:
In that case, instead of using len(ret['keys']) != 0,  the add_to_tables_not_found argument could be set to False could set the `add_to_ret_template(self, table, db, keys, err, add_to_tables_not_found=True)`